src/DataTypes/DataTypesNumber.cpp

```diff
@@ -87,16 +87,18 @@ void registerDataTypeNumbers(DataTypeFactory & factory)
     factory.registerAlias("MEDIUMINT SIGNED", "Int32", DataTypeFactory::CaseInsensitive);
     factory.registerAlias("INT SIGNED", "Int32", DataTypeFactory::CaseInsensitive);
     factory.registerAlias("INTEGER SIGNED", "Int32", DataTypeFactory::CaseInsensitive);
+    factory.registerAlias("SIGNED", "Int64", DataTypeFactory::CaseInsensitive);
```

Comment:
Let's update the docs at https://clickhouse.com/docs/en/sql-reference/data-types/int-uint

Reply:
I know all of this is motivated by log files but did you find a description in MySQLs [docs](https://dev.mysql.com/doc/refman/8.0/en/numeric-type-syntax.html) about `SIGNED`/`UNSIGNED`? (I didn't)

Just asking to avoid surprises, e.g. because (hypothetically) the integer width depends on settings, on the platform, the OS or whatever funny things.

Reply:
It is from this one: https://dev.mysql.com/doc/refman/8.0/en/cast-functions.html#function_cast

If you scroll a fair amount, there is an entry:

![image](https://github.com/ClickHouse/ClickHouse/assets/3175289/6e77e21f-4180-48cb-a611-32df803e1f31)

This is inconsistent with the table definition syntax (INTEGER UNSIGNED = UInt32, CAST(x AS UNSIGNED INTEGER = UInt64).

Reply:
Alright, thanks.

Well, I guess it all makes "sense" if one consers `INTEGER UNSIGNED` and `UNSIGNED INTEGER` as unrelated types. Since table DDL syntax doesn' have `UNSIGNED` or `SIGNED` types, there is also no risk of ambiguities in our emulation of MySQL's behavior.

Let's add the aliases and not worry too much.

Reply:
@rschu1ze, since this change will be global, it is somewhat confusing and is supposed to work only in MySQL's CAST; let's maybe not add it in this PR? If it will required in the future, I'd instead add a CAST overload for MySQL only where we can encapsulate all the SIGNED/UNSIGNED awkwardness (and remove some aliases from the globals).

Reply:
Sounds like a plan.